### ISSUE TITLE
Feature request - Block known numbers

### ISSUE THREAD

**Issue by Ilithy** [x13a/Silence#141]
Hi, I have a feature request (a bit special), it would be an option in Silence (like the main elements, Message, repeated etc...)
- that would block all calls from numbers known by the device (including numbers found in messages, numbers called without being in the contacts etc...).
- When enabled, it will disable all authorization options of the main buttons (which have been toggled, and restore their state when the option is disabled.

this would create an airplane mode without disconnecting from the network which would be very practical when you want to be quiet.

Thank you.

**Comment by x13a** [comment]
Hi,

When it enabled only contacts will be allowed to call you. Am I right?

**Comment by Ilithy** [comment]
Greetings,
No, on the contrary, this option would prevent calls from numbers known by the phone (including contacts, numbers contacted without being in the contacts, numbers from sms etc...)

**Comment by x13a** [comment]
And what about numbers not known by the phone?

**Comment by Ilithy** [comment]
Well either they are blocked too, or there could be a sub-parameters that will only allow calls from these unknown phone numbers (excluding hidden numbers).

**Comment by x13a** [comment]
Does not DND better for this?

**Comment by Ilithy** [comment]
Indeed, the do not disturb mode is very practical, but it does not allow the fact:
to authorize only calls from unknown numbers, it only allows to authorize calls from contacts, or contacts in favorites.
and it does not allow the management of the rejection mode of the call (block, refuse or silence).

**Comment by x13a** [comment]
Got it)

**Comment by Ilithy** [comment]
thank you so much

**Comment by Ilithy** [comment]
The new option works very well, and prevents contacts from calling us, thanks a lots.

**Comment by x13a** [comment]
I thought you want more complex thing..

**Comment by Ilithy** [comment]
Indeed it was my initial idea (block contacts, and numbers known but not registered as contacts), see even a tile managing the activation / deactivation of the feature, but I did not think you would want to implement everything (it is a lot of work I know)

**Comment by x13a** [comment]
@Ilithy have added block mode, but global for now

**Comment by Ilithy** [comment]
Thank you so much, I can't wait to see this, thanks again <3
